File: LeetCode/1481/solution.java
// 1481. Least Number of Unique Integers after K Removals
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int a: arr) {
            map.put(a, map.getOrDefault(a, 0)+1);
        }
        
        List<Integer> v = new ArrayList<>();
        for(int a: map.values()) {
            v.add(a);
        }

        Collections.sort(v);

        int cnt = 0;
        for(int a: v){
            if(k < a) break;
            k -= a;
            cnt++;
        }

        return v.size()-cnt;
    }
}

File: LeetCode/1481/solution.cpp
// 1481. Least Number of Unique Integers after K Removals
class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<int, int> freq;
        for(auto a: arr) freq[a]++;
        
        vector<int> v;
        for(auto a: freq) {
            v.push_back(a.second);
        }

        sort(v.begin(), v.end());

        int cnt = 0;
        for(auto a: v){
            if(k < a) break;
            k -= a;
            cnt++;
        }

        return v.size()-cnt;
    }
};